### PR TITLE
clarify error message for root user credential

### PR DESCRIPTION
**Pull request by austin880625**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description

Clarifies error messages for root user credential environment variables so that it matches the actual failure reason.

## Motivation and Context

MINIO_ACCESS_KEY and MINIO_SECRET_KEY is prompted as deprecated when set in enviroment variables. The error message on failure of credential check still shows "access key" and "secret key", which confuses users. 

## How to test this PR?

start minio server with MINIO_ROOT_USER and MINIO_ROOT_PASSWORD set and violate the length requirements

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
This PR needs to be simplified its changing too many things.

**Comment by austin880625** [comment]
Sorry I'm new to this project. I just pushed another version which only changes the error displayed when `MINIO_ROOT_USER` or `MINIO_ROOT_PASSWORD` doesn't meet the credential requirements(removed the changes for env file credentials). Is it still too much for this PR?

File: cmd/common-main.go [hunk 1]
```diff
@@ -843,6 +843,7 @@ func loadRootCredentials() {
 	// Check both cases and authenticate them if correctly defined
 	var user, password string
 	var hasCredentials bool
+	isUsingKeys := false
```
Comment (harshavardhana) [inline cmd/common-main.go]:
```suggestion
	var legacyCredentials bool
```

File: cmd/common-main.go [hunk 2]
```diff
@@ -851,6 +852,7 @@ func loadRootCredentials() {
 	} else if env.IsSet(config.EnvAccessKey) && env.IsSet(config.EnvSecretKey) {
 		user = env.Get(config.EnvAccessKey, "")
 		password = env.Get(config.EnvSecretKey, "")
+		isUsingKeys = true
```
Comment (harshavardhana) [inline cmd/common-main.go]:
```suggestion
		legacyCredentials = true
```

File: cmd/common-main.go [hunk 3]
```diff
@@ -859,8 +861,13 @@ func loadRootCredentials() {
 	if hasCredentials {
 		cred, err := auth.CreateCredentials(user, password)
 		if err != nil {
-			logger.Fatal(config.ErrInvalidCredentials(err),
-				"Unable to validate credentials inherited from the shell environment")
+			if isUsingKeys {
```
Comment (harshavardhana) [inline cmd/common-main.go]:
```suggestion
			if legacyCredentials {
```

**Comment by harshavardhana** [comment]
the new PR looks good.

**Comment by austin880625** [comment]
I've just updated the naming. Thank for the review

**Review by harshavardhana — APPROVED** [review]

**Comment by austin880625** [comment]
Oops sorry I was using the phone and touched the "merge master" button unintentionally, should we restart the pipeline or I need to force push again?

**Pull request merged**